CMakeLists.txt

```diff
@@ -28,6 +28,20 @@ cmake_minimum_required(VERSION 3.17)
 
 project(tritoncommon LANGUAGES C CXX)
 
+if(CMAKE_CXX_COMPILER_ID STREQUAL "MSVC")
+  message("Using MSVC as compiler, default target on Windows 10. "
+		  "If the target system is not Windows 10, please update _WIN32_WINNT "
+		  "to corresponding value.")
+  set(CMAKE_CXX_FLAGS "/W4 /D_WIN32_WINNT=0x0A00 /EHsc")
```

Comment:
Don't use global cmake vars. if-else the target_compile_options below to set CMAKE_CXX_FLAGS in the same way as doing it for non-windows below. Don't need any of these other settings as they are already covered below or I put a note of how to add them for the target below.